tests/test_math_kvecgenerator.cc

```diff
@@ -0,0 +1,280 @@
+ *        of the spherical expansion coefficients of SOAP and LODE
+ *
+ * Copyright  2019  Felix Musil, COSMO (EPFL), LAMMM (EPFL)
+ *
+ * Rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * Rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include "test_math.hh"
+
+#include "rascal/math/kvec_generator.hh"
+
+#include <boost/test/unit_test.hpp>
+
+#include <cmath>
+
+namespace rascal {
+
+  BOOST_AUTO_TEST_SUITE(test_kvecgenerator);
+
+  /* ---------------------------------------------------------------------- */
+  /**
+   * Test that the kvecgenerator produces the desired vectors for small
+   * cells, where we can explicitly list the correct vectors
+   */
+  BOOST_AUTO_TEST_CASE(kvecgen_smallcells_test) {
+    // Define reciprocal space cell to be used in the following tests
+    // We use the identity matrix to avoid floating point errors
+    Eigen::Matrix3d reciprocalcell;
+    reciprocalcell << 1, 0, 0, 0, 1, 0, 0, 0, 1;
+
+    // Store vectors that should be found, if the code runs correctly
+    Eigen::RowVector3d v100, v010, v001, v110, v101, v011, v01m, v10m, v1m0,
+        v111, v11m, v1m1, v1mm;
+    v100 << 1, 0, 0;
+    v010 << 0, 1, 0;
+    v001 << 0, 0, 1;
+    v110 << 1, 1, 0;
+    v101 << 1, 0, 1;
+    v011 << 0, 1, 1;
+    v01m << 0, 1, -1;
+    v10m << 1, 0, -1;
+    v1m0 << 1, -1, 0;
+    v111 << 1, 1, 1;
+    v11m << 1, 1, -1;
+    v1m1 << 1, -1, 1;
+    v1mm << 1, -1, -1;
+
+    // For small enough cutoff radii, namely for the intervals
+    // [1,sqrt(2)), [sqrt(2), sqrt(3)), [sqrt(3),2) we can explicitly
+    // list all vector that should be found by the algorithm.
+    // Store these solutions in the correct order for each cutoff
+    Eigen::MatrixXd kvec_sol0(3, 3), kvec_sol1(9, 3), kvec_sol2(13, 3);
+    kvec_sol0 << v001, v010, v100;
+    kvec_sol1 << v001, v01m, v010, v011, v1m0, v10m, v100, v101, v110;
+    kvec_sol2 << v001, v01m, v010, v011, v1mm, v1m0, v1m1, v10m, v100, v101,
+        v11m, v110, v111;
+    Eigen::MatrixXd * solutions[3] = {&kvec_sol0, &kvec_sol1, &kvec_sol2};
+
+    // For various cutoff radii close to the thresholds after which
+    // we would get further vectors, we compare the obtained vectors
+    // to the correct solution. The variable i runs over the three intervals
+    double cutofflims[4] = {1., math::SQRT_TWO, math::SQRT_THREE, 2.};
```

Comment:
You should use underscores to separate words in your variables names, it makes it much easier to read the code

```suggestion
    double cutoff_lims[4] = {1., math::SQRT_TWO, math::SQRT_THREE, 2.};
```